src/psyclone/transformations.py

```diff
@@ -461,55 +617,74 @@ def apply(self, node, reprod=None):
         with the same number of OpenMP threads, not for different
         numbers of OpenMP threads.
 
-        :param node: the supplied node to which we will apply the
+        :param node: the supplied node to which we will apply the \
         OMPLoopTrans transformation
         :type node: :py:class:`psyclone.psyGen.Node`
-        :param reprod: optional argument to determine whether to
-        generate reproducible OpenMP reductions (True) or not
-        (False). The default value is None which will cause PSyclone
+        :param reprod: optional argument to determine whether to \
+        generate reproducible OpenMP reductions (True) or not \
+        (False). The default value is None which will cause PSyclone \
         to look up a default value
-        :type reprod: Boolean, or None
-        :return: (:py:class:`psyclone.psyGen.Schedule`,
+        :type reprod: Boolean or None
+        :return: (:py:class:`psyclone.psyGen.Schedule`, \
         :py:class:`psyclone.undoredo.Memento`)
 
         '''
+        if reprod:
+            self._reprod = reprod
 
-        self._validate(node)
-
-        if reprod is None:
-            import psyclone.config
-            reprod = psyclone.config.REPRODUCIBLE_REDUCTIONS
-
-        schedule = node.root
-
-        # create a memento of the schedule and the proposed
-        # transformation
-        from psyclone.undoredo import Memento
-        keep = Memento(schedule, self, [node])
+        return super(OMPLoopTrans, self).apply(node)
 
-        # keep a reference to the node's original parent and its index as these
-        # are required and will change when we change the node's location
-        node_parent = node.parent
-        node_position = node.position
 
-        # add our orphan OpenMP loop directive setting its parent to
-        # the node's parent and its children to the node
-        from psyclone.psyGen import OMPDoDirective
-        directive = OMPDoDirective(parent=node_parent,
-                                   children=[node],
-                                   omp_schedule=self.omp_schedule,
-                                   reprod=reprod)
+class ACCLoopTrans(ParallelLoopTrans):
+    '''
+    Adds an OpenACC loop directive to a loop. This directive must be within
+    the scope of some OpenACC Parallel region.
+    '''
```

Comment:
This needs an example in the description like the other OpenACC directives (see generated documentation)

Reply:
Done.